pkg/scheduler/api/podgroup_info/subgroup_info.go

```diff
@@ -0,0 +1,31 @@
+// Copyright 2025 NVIDIA CORPORATION
+// SPDX-License-Identifier: Apache-2.0
+
+package podgroup_info
+
+import (
+	"github.com/NVIDIA/KAI-scheduler/pkg/apis/scheduling/v2alpha2"
+	"github.com/NVIDIA/KAI-scheduler/pkg/scheduler/api/pod_info"
+)
+
+type SubGroupInfo struct {
+	Name         string
+	MinAvailable int32
+	PodInfos     pod_info.PodsMap
+}
+
+func NewSubGroupInfo(name string, minAvailable int32) *SubGroupInfo {
+	return &SubGroupInfo{
+		Name:         name,
+		MinAvailable: minAvailable,
+		PodInfos:     pod_info.PodsMap{},
+	}
+}
+
+func FromSubGroup(subGroup *v2alpha2.SubGroup) *SubGroupInfo {
+	return NewSubGroupInfo(subGroup.Name, subGroup.MinMember)
+}
+
+func (sgi *SubGroupInfo) AddTaskInfo(ti *pod_info.PodInfo) {
```

Comment:
What about "removeTaskInfo"?

Reply:
Shouldn't this method be used by the jobInfo?

Reply:
You are right, it probably will be more aligned this way.

While looking into it I noticed that the only use case for deleting a task is when task status is updated, old task is removed and new task, with the new status is added. And when it happens we don't delete the old task from PodInfos map - the old task is cloned and `Deleted` status is set.
https://github.com/NVIDIA/KAI-Scheduler/blob/daf2d15cf3c5266efe231108fa47c53ade426590/pkg/scheduler/api/podgroup_info/job_info.go#L265
Right after the task is cloned it is overwritten with the new PodInfo that has the same UID:
https://github.com/NVIDIA/KAI-Scheduler/blob/daf2d15cf3c5266efe231108fa47c53ade426590/pkg/scheduler/api/podgroup_info/job_info.go#L189

Isn't it redundant?

Reply:
Unless there is a good reason for that, I can change the code so deleted task will not clone the task and instead - it will be removed from PodInfos map.

Reply:
I think that in some cases there are some pointers to the unupdated task. You can try and edit this, but please make a full e2e run with real gpus and see if something breaks

Reply:
After speaking about it, we decided to do a small refactoring in this area:
https://github.com/NVIDIA/KAI-Scheduler/pull/345